File: crates/aft/tests/fixtures/sample.go
// Sample Go file for tree-sitter symbol extraction tests.
package sample

import "fmt"

// ExportedFunction is a public function (uppercase).
func ExportedFunction(x int) int {
	return x + 1
}

// unexportedFunction is a private function (lowercase).
func unexportedFunction() {
	fmt.Println("private")
}

// MyStruct is a sample struct type.
type MyStruct struct {
	Name string
	Age  int
}

// Reader is a sample interface type.
type Reader interface {
	Read(p []byte) (n int, err error)
}

// String is a method with receiver on MyStruct.
func (m *MyStruct) String() string {
	return m.Name
}

// helper is an unexported method.
func (m *MyStruct) helper() int {
	return m.Age
}

File: crates/aft/tests/fixtures/sample.rs
/// Sample Rust module for tree-sitter symbol extraction tests.

pub fn public_function(x: i32) -> i32 {
    x + 1
}

fn private_function() {
    // not exported
}

pub struct MyStruct {
    pub field: String,
}

pub enum Color {
    Red,
    Green,
    Blue,
}

pub trait Drawable {
    fn draw(&self);
    fn area(&self) -> f64;
}

impl MyStruct {
    pub fn new(field: String) -> Self {
        Self { field }
    }

    fn helper(&self) -> usize {
        self.field.len()
    }
}

impl Drawable for MyStruct {
    fn draw(&self) {
        println!("drawing {}", self.field);
    }

    fn area(&self) -> f64 {
        0.0
    }
}
